tools/adios2pio-nm/adios2pio-nm-lib.cxx

```diff
@@ -1379,201 +1421,82 @@ int adios2_ConvertVariableTimedPutVar(adios2::Variable<T> *v_base, std::vector<T
                      << endl;
         }
 
-        /* Is this a local array written by each process, or a truly distributed global array */
         TimerStart(read);
 
         *v_base = bpIO[0].InquireVariable<T>(varname.c_str());
         const auto v_blocks = bpReader[0].BlocksInfo(*v_base, 0);
 
         TimerStop(read);
 
-        bool local_array = true;
-        typename adios2::Variable<T>::Info v_info = v_blocks[0];
-        adios2::Dims b_dims = v_info.Count;
-        for (size_t d = 0; d < v_dims.size(); d++)
-        {
-            if (b_dims[d] != v_dims[d])
-            {
-                local_array = false;
-                break;
-            }
-        }
-        if (var.nctype == PIO_CHAR && v_dims.size() == 1)
-        {
-            /* Character array over time may have longer dimension declaration than the actual content */
-            local_array = true;
-        }
+        char start_varname[PIO_MAX_NAME];
+        char count_varname[PIO_MAX_NAME];
+        sprintf(start_varname, "start_id/%s", varname.c_str());
```

Comment:
Can we use snprintf here too?

Reply:
Shall I convert sprintf to snprintf and push the changes or would you like @dqwu to do it?

Reply:
@tkurc I will make the changes.